library/L2_HAL/communication/test/tsop752_test.cpp

```diff
@@ -0,0 +1,143 @@
+#include "L2_HAL/communication/tsop752.hpp"
+#include "L4_Testing/testing_frameworks.hpp"
+
+namespace sjsu
+{
+EMIT_ALL_METHODS(Tsop752);
+
+TEST_CASE("Tsop752 Infrared Receiver Test", "[tsop752]")
+{
+  Mock<PulseCapture> mock_pulse_capture;
+  Mock<Timer> mock_timer;
+
+  Fake(Method(mock_pulse_capture, ConfigureCapture),
+       Method(mock_pulse_capture, EnableCaptureInterrupt));
+  Fake(Method(mock_timer, SetMatchBehavior),
+       Method(mock_timer, Start),
+       Method(mock_timer, Stop),
+       Method(mock_timer, Reset));
+
+  const Status kFailureStatus[] = {
+    Status::kTimedOut,          Status::kBusError,
+    Status::kDeviceNotFound,    Status::kInvalidSettings,
+    Status::kNotImplemented,    Status::kNotReadyYet,
+    Status::kInvalidParameters,
+  };
+  constexpr std::chrono::microseconds kTimeout = 5ms;
+
+  Tsop752 ir_receiver(mock_pulse_capture.get(), mock_timer.get(), kTimeout);
+
+  SECTION("Initialize")
+  {
+    SECTION("PulseCapture peripheral initialization failure")
+    {
+      mock_timer.ClearInvocationHistory();
+      When(Method(mock_timer, Initialize)).AlwaysReturn(Status::kSuccess);
+      for (uint8_t i = 0; i < std::size(kFailureStatus); i++)
```

Comment:
```suggestion
      for (size_t i = 0; i < kFailureStatus.size(); i++)
```
Use the size_t for the rest of the for loops here.